### ISSUE TITLE
Capture rails runner's exceptions

### ISSUE THREAD

**Issue by manuelmeurer** [getsentry/sentry-ruby#1789]
### Issue Description

When I run `bin/rails runner "raise 'test error'"`, it is not sent to Sentry. (same with `bundle exec rails runner`).
I was under the impression that this was fixed in #1448.

`bin/rails runner "Sentry.capture_message 'text'"` and `bin/rails runner "Sentry.capture_exception StandardError.new('text')"` both work as expected.
`bin/rails runner "puts Sentry.configuration.background_worker_threads"` prints `0`, also as expected.
What am I doing wrong?

### Reproduction Steps

Run `bin/rails runner "raise 'test error'"`

### Expected Behavior

An error reported to Sentry

### Actual Behavior

No error is reported to Sentry

### Ruby Version

3.1.1

### SDK Version

5.2.1

### Integration and Its Version

Rails 5.2.1, Sidekiq 5.2.1

### Sentry Config

```
Sentry.init do |config|
  config.breadcrumbs_logger             = %i(active_support_logger http_logger)
  config.traces_sample_rate             = 0.2
  config.capture_exception_frame_locals = true
  config.release                        = Rails.configuration.revision
  config.send_default_pii               = true
end
```

**Comment by st0012** [comment]
That's because `sentry-rails` currently doesn't capture exceptions from Rails `console` or `runner` command. I think I would still not support `console` because it's too easy to trigger events because of typos. But I think `runner`'s case may be debatable. 

I personally also use `rails runner` for debugging/experimenting purpose more so I don't usually want to report its exceptions automatically.

**Comment by manuelmeurer** [comment]
Thanks for the clarification, @st0012!

I agree that errors in `console` should not be captured, since the user sees them immediately after typing in.
When using `runner`, my expectation was that errors are captured by default, though.
In our project (and in many other projects I worked on), `runner` is used together with Cron to execute hourly, daily or weekly tasks, i.e. `bundle exec rails runner Tasks::Hourly.call`.

If other people use `runner` more for experimentation like you as well, maybe a configuration option that defaults to false (`config.capture_runner_errors = false`) would be possible?

**Comment by manuelmeurer** [comment]
What would be a way to explicitly tell Sentry to capture errors when using `runner` for now?
I thought `bundle exec rails runner "Sentry.with_session_tracking { raise 'test' }"` would work but it doesn't...

**Comment by st0012** [comment]
`Sentry.with_session_tracking` only changes the session stat sent to Sentry. It doesn't capture exceptions unfortunately.

I just checked [rails runner's doc](https://guides.rubyonrails.org/command_line.html#bin-rails-runner) and it looks like running production code/script is frequent usage, so I'll support exception capturing for it.

**Comment by st0012** [comment]
Unfortunately, there's no API for providing such integration at the moment. I have [proposed a new API](https://discuss.rubyonrails.org/t/proposal-provide-api-to-wrap-runner-commands-execution/80468) for this purpose but I'm not sure if it'd be accepted. And even if it were accepted, it will only be implemented in Rails 7.1 and won't help your case much 😬 

I'm considering using the `at_exit` workaround as `Airbrake` does in (https://github.com/airbrake/airbrake/pull/585).

Also, I think having a `Sentry.with_exception_captured { my_code }` to capture exceptions would be nice. (`Sentry.capture_exception { block }` has been taken for customizing the event). 

@sl0thentr0py Wdyt?

**Comment by sl0thentr0py** [comment]
yea I'm fine with both `at_exit` and a new wrapper for exception capturing.
